classes/Commands/UpdateCommand.php

```diff
@@ -0,0 +1,191 @@
+            ->addOption('chain', 'ca', InputOption::VALUE_NONE, 'Allows you to chain update commands.')
+            ->addOption('channel', 'ce', InputOption::VALUE_OPTIONAL, 'Selects what update to run (minor, major etc.)')
+            ->addOption('config-file-path', 'cf', InputOption::VALUE_OPTIONAL, 'Configuration file location for update.')
+            ->addOption('action', 'a', InputOption::VALUE_OPTIONAL, 'Advanced users only. Sets the step you want to start from (Default: UpgradeNow, see https://devdocs.prestashop-project.org/8/basics/keeping-up-to-date/upgrade-module/upgrade-cli/ for other values available)')
+            ->addOption('data', 'd', InputOption::VALUE_OPTIONAL, 'Advanced users only. Contains the state of the update process encoded in base64');
+    }
+
+    /**
+     * @throws Exception
+     */
+    protected function execute(InputInterface $input, OutputInterface $output): ?int
+    {
+        $this->logger = $output->isDecorated() ? new CliLogger($output) : new StreamedLogger();
+        if ($output->isQuiet()) {
+            $this->logger->setFilter(Logger::ERROR);
+        } elseif ($output->isVerbose()) {
+            $this->logger->setFilter(Logger::DEBUG);
+        } else {
+            $this->logger->setFilter(Logger::INFO);
+        }
+
+        $this->logger->debug('Starting the update process.');
+
+        try {
+            $prodRootDir = _PS_ROOT_DIR_;
+            $this->logger->debug('Production root directory: ' . $prodRootDir);
+
+            $adminDir = $input->getOption('admin-dir');
+            $this->logger->debug('Admin directory: ' . $adminDir);
+            define('_PS_ADMIN_DIR_', _PS_ROOT_DIR_ . DIRECTORY_SEPARATOR . $adminDir);
+
+            $upgradeContainer = new UpgradeContainer($prodRootDir, $adminDir);
+            $this->logger->debug('Upgrade container initialized.');
+
+            $this->logger->debug('Logger initialized: ' . get_class($this->logger));
+
+            $upgradeContainer->setLogger($this->logger);
+            (new ErrorHandler($this->logger))->enable();
+            $this->logger->debug('Error handler enabled.');
+
+            $configPath = $input->getOption('config-file-path');
+            if (!empty($configPath)) {
+                $exitCode = $this->loadConfiguration($configPath, $upgradeContainer);
+                if ($exitCode !== ExitCode::SUCCESS) {
+                    return $exitCode;
+                }
+                $this->logger->debug('Configuration loaded successfully.');
+            }
+
+            $controller = new AllUpgradeTasks($upgradeContainer);
+            $controller->setOptions([
+            'data' => $input->getOption('data'),
+            'action' => $input->getOption('action'),
+            'channel' => $input->getOption('channel'),
+        ]);
+            $controller->init();
+            $exitCode = $controller->run();
+            $this->logger->debug('Controller run completed with exit code: ' . $exitCode);
```

Comment:
Is "Controller" the proper wording? I don't know if this is applicable in a CLI context

```suggestion
            $this->logger->debug('Process completed with exit code: ' . $exitCode);
```